py/picca/pk1d/postproc_pk1d.py

```diff
@@ -922,39 +906,79 @@ def compute_cov(
             k1_array,
             k2_array,
         )
-
-    for sub_forest_id in sub_forest_ids:  # First loop 1) id sub-forest bins
+    # First loop 1) id sub-forest bins
+    for sub_forest_id in sub_forest_ids:
         select_id = select_z & (p1d_table["sub_forest_id"] == sub_forest_id)
         selected_pk = p1d_table["Pk"][select_id]
         selected_ikbin = k_index[select_id]
 
-        for ipk, _ in enumerate(selected_pk):  # First loop 2) selected pk
-            ikbin = selected_ikbin[ipk]
-            if ikbin != -1:
-                for ipk2 in range(ipk, len(selected_pk)):  # First loop 3) selected pk
+        if weight_method == "fit_snr":
+            # Definition of weighted unbiased sample covariance taken from
+            # "George R. Price, Ann. Hum. Genet., Lond, pp485-490,
+            # Extension of covariance selection mathematics, 1972"
+            # adapted with weights w_i = np.sqrt(w_ipk * w_ipk2) for covariance
+            # to obtain the right definition of variance on the diagonal
+            selected_snr = p1d_table["forest_snr"][select_id]
+            snrfit_z = snrfit_table[izbin * nbins_k : (izbin + 1) * nbins_k, :]
+            selected_variance_estimated = fitfunc_variance_pk1d(
+                selected_snr, snrfit_z[selected_ikbin, 2], snrfit_z[selected_ikbin, 3]
+            )
+            # First loop 2) selected pk
+            for ipk, _ in enumerate(selected_pk):
+                ikbin = selected_ikbin[ipk]
+                # First loop 3) selected pk
+                for ipk2 in range(ipk, len(selected_pk)):
                     ikbin2 = selected_ikbin[ipk2]
-
-                    if ikbin2 != -1:
+                    if (ikbin2 != -1) & (ikbin != -1):
+                        # index of the (ikbin,ikbin2) coefficient on the top of the matrix
+                        index = (nbins_k * ikbin) + ikbin2
+                        weight = 1 / selected_variance_estimated[ipk]
+                        weight2 = 1 / selected_variance_estimated[ipk2]
+                        covariance_array[index] = covariance_array[index] + selected_pk[
+                            ipk
+                        ] * selected_pk[ipk2] * np.sqrt(weight * weight2)
```

Comment:
why is the weighting done with a sqrt(weight *weight2) instead of with the weights directly, see other comment